### DISCUSSION TITLE
让你的简悦客户端变成SaaS服务，低代码Workflow工具Automa增强简悦使用体验

### DISCUSSION THREAD

**Discussion by shadu120** [Kenshin/simpread#4913] [服务]
你好！相信你一定有这样的体验：
- 看到一篇不错的文章，想把他快速收藏起来，回头再看。但是等回了头，发现文章删除了……
- 一些常见的网址保存工具只能帮你存地址，而内容却无能为力，即使能保存，也是丑得一塌糊涂……
-  很庆幸，你找到了简悦！她能帮你一键美化网页，并存储为离线markdown、html和pdf，但是需要手动。我这人在路上，怎么操作简悦呢？
-  是不是你也想过：要是简悦客户端的功能一下子变成SaaS就好了！

**简悦为了安全和用户隐私，十分克制自己的功能，所以目前还没有SaaS化，不是不能，是有所克制。这是我理解**

# 不啰嗦了，操作吧！

## 第三步：从Automa官方应用市场导入Workflow[SimpRead.Auto](https://www.automa.site/workflow/4oeVbmrLr-JQgpnKnmP21)
<img width="1600" alt="image" src="https://user-images.githubusercontent.com/3023471/207820075-ee8ceecc-7ee2-4bbc-824b-7c9296b159e6.png">

<img width="496" alt="image" src="https://user-images.githubusercontent.com/3023471/207819950-1ccf4a72-1e8f-489c-b6e0-d57f28c42457.png">

## 第四步：修改Workflow全局变量
<img width="1726" alt="image" src="https://user-images.githubusercontent.com/3023471/207820276-d062da35-88a1-45a6-8ab1-b64a7e4be5c6.png">

<img width="1336" alt="image" src="https://user-images.githubusercontent.com/3023471/207820456-c75c3eef-0460-4f72-b83b-98737b539434.png">

这个URL： https://YOUR-CUSTOMER-DOMAIN/json_article_to_save 需要你自己实现，返回内容格式，请参考workflow里的notes

修改后，保存一下Workflow

## 第五步：设置一个定时任务，让Workflow每分钟运行一次
<img width="1725" alt="image" src="https://user-images.githubusercontent.com/3023471/207820976-2048da22-7668-4738-b4fe-8c934542701e.png">

## 如果你想保存导出记录（可选）
要想针对所有历史上存过的文章进行数据保存，可以在Automa的Storage中创建一个表。
<img width="1709" alt="image" src="https://user-images.githubusercontent.com/3023471/207822691-b80897e0-91a8-412c-bf77-c295af709881.png">

把这个表和workflow进行连接
<img width="1236" alt="image" src="https://user-images.githubusercontent.com/3023471/207822998-05fc058a-6bdc-4929-b8fb-cc85f797394b.png">

# 大致效果

- 你电脑/手机看到一篇文章不错，把这个地址发给某个TG机器人或其他系统…（自己实现）
- 1分钟后，SimpRead.Auto会拉取到这个文章的URL
- Automa会用浏览器新标签页打开这个URL，并依次执行快捷键[aa]--[sr]--[rl]--[pf]，阅读模式--快捷键---存入稍后读--导出为本地pdf
- Automa会在完成所有工作后，关闭浏览器标签
- 你在简悦客户端output文件夹下可以看到一个新增的pdf文件

# 还可以继续
- 把这个已经获得的PDF文件，上传到dropbox
- 获取一个dropbox的分享连接
- 转发到TG，完美！

借鉴了这个玩法：[基于简悦的 API 实现的 Telegram bot 可以方便的将任何 URL 发送到简悦的稍后读中。](https://github.com/Kenshin/simpread/discussions/2792)

**Comment by shadu120** [comment]
后续效果：

<img width="1195" alt="image" src="https://user-images.githubusercontent.com/3023471/211268658-affc1bfc-bdda-4ea5-a8b4-70af2e70d747.png">

**Reply by Kenshin** [reply]
Cool~  💪  🎉

**Comment by kbrx93** [comment]
cool !😎
